src/legacy/core_plugins/data/public/filter/filter_service.ts

```diff
@@ -29,22 +27,15 @@ import { FilterManager } from './filter_manager';
 
 export interface FilterServiceDependencies {
   indexPatterns: IndexPatterns;
+  uiSettings: UiSettingsClientContract;
 }
 
 export class FilterService {
-  public setup({ indexPatterns }: FilterServiceDependencies) {
-    const filterManager = new FilterManager(indexPatterns);
+  public setup({ indexPatterns, uiSettings }: FilterServiceDependencies) {
+    const filterManager = new FilterManager(indexPatterns, uiSettings);
 
     return {
       filterManager,
-      ui: {
-        ApplyFiltersPopover,
-        FilterBar,
-      },
-      loadLegacyDirectives: once(() => {
```

Comment:
I have mixed feelings about changing this approach; I like the clarity provided by a dedicated `legacy` module because it makes life easier for us as owners of the data plugin.

But the tradeoff is that we're no longer being quite as explicit about what's being loaded when we do `import 'data/public/legacy'`, and we no longer have directives isolated to an individual service.

So for downstream developers, they now have a global set of legacy data directives instead of ones scoped to an individual service. This might make their migration path a bit more tricky and/or load more directives than they truly need.